### PR TITLE
Update test-framework submodule

### PR DESCRIPTION
**Pull request by robertbaldyga**
- Remove deprecated TestRun.exception()
- Use TestRun.makereport()

### CODE REVIEW

**Comment by robertbaldyga** [comment]
> import CmdException in installer.py

Done.

File: test/functional/api/cas/installer.py [hunk 1]
```diff
@@ -23,21 +24,21 @@ def install_opencas():
         "./configure && "
         "make -j")
     if output.exit_code != 0:
-        TestRun.exception(
+        raise CmdException(
             f"Make command executed with nonzero status: {output.stdout}\n{output.stderr}")
```
Comment (Deixx) [inline test/functional/api/cas/installer.py]: pass output as second arg to CmdException and remove stdout and stderr from f-string

File: test/functional/api/cas/installer.py [hunk 2]
```diff
@@ -23,21 +24,21 @@ def install_opencas():
...
 
     TestRun.LOGGER.info("Installing Open CAS")
     output = TestRun.executor.run(
         f"cd {TestRun.plugins['opencas'].working_dir} && "
         f"make install")
     if output.exit_code != 0:
-        TestRun.exception(
+        raise CmdException(
             f"Error while installing Open CAS: {output.stdout}\n{output.stderr}")
```
Comment (Deixx) [inline test/functional/api/cas/installer.py]: same here and in the following CmdExceptions

Comment (robertbaldyga) [reply test/functional/api/cas/installer.py]: Done.

File: test/functional/api/cas/installer.py (hunk 1 above)
Comment (robertbaldyga) [reply test/functional/api/cas/installer.py]: Done.

**Pull request merged**